warden/x/async/keeper/futures.go

```diff
@@ -0,0 +1,77 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/collections"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/warden-protocol/wardenprotocol/warden/repo"
+	types "github.com/warden-protocol/wardenprotocol/warden/x/async/types/v1beta1"
+)
+
+type FutureKeeper struct {
+	futures         repo.SeqCollection[types.Future]
+	futureByCreator collections.KeySet[collections.Pair[sdk.AccAddress, uint64]]
+	results         collections.Map[uint64, types.FutureResult]
+}
+
+func NewFutureKeeper(sb *collections.SchemaBuilder, cdc codec.Codec) *FutureKeeper {
+	futuresSeq := collections.NewSequence(sb, FuturesPrefix, "futures_sequence")
+	futuresColl := collections.NewMap(sb, FutureSeqPrefix, "futures", collections.Uint64Key, codec.CollValue[types.Future](cdc))
+
+	futures := repo.NewSeqCollection(futuresSeq, futuresColl, func(t *types.Future, u uint64) { t.Id = u })
+	futureByCreator := collections.NewKeySet(sb, FutureByAddressPrefix, "futures_by_address", collections.PairKeyCodec(sdk.AccAddressKey, collections.Uint64Key))
+
+	results := collections.NewMap(sb, ResultsPrefix, "future_results", collections.Uint64Key, codec.CollValue[types.FutureResult](cdc))
+
+	return &FutureKeeper{
+		futures:         futures,
+		futureByCreator: futureByCreator,
+		results:         results,
+	}
+}
+
+func (k *FutureKeeper) Append(ctx context.Context, t *types.Future) (uint64, error) {
```

Comment:
Do you think the name `append` is okay here? Wouldn't it be better to name it something like `Schedule`, to make it less general?

Reply:
Personally, I think Append is fine, we always used it in keepers to indicate that you are creating a new element in a collection

how this new item is going to be used is not really a concern of this function